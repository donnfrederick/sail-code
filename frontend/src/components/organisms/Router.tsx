import * as Sentry from '@sentry/browser'
import ErrorPage from 'components/pages/Error'
import { createBrowserHistory } from 'history'
import * as React from 'react'
import { Provider, Store } from 'react-redux'
import { ConnectedRouter } from 'react-router-redux'
import { compose, lifecycle, withState } from 'recompose'
import createStore from 'store/createStore'
import resetViewport from 'utils/resetViewport'
import userAgent from 'utils/userAgent'

declare const gtag: any

export const history = createBrowserHistory()
export const store = createStore(history) as Store<any>

history.listen(() => {
  window.scrollTo(0, 0)
  gtag('config', 'UA-105776977-2', { page_path: history.location.pathname })
  userAgent()
  resetViewport()
})

interface Props {
  children: JSX.Element
  errors: Error | null
  setErrors: (state: any) => any
}

const isProd = process.env.NODE_ENV === 'production'

const sentry = compose(
  withState('errors', 'setErrors', null),
  isProd
    ? lifecycle<Props, {}, {}>({
        componentDidCatch(error, errorInfo) {
          this.props.setErrors(error)
          Sentry.withScope(scope => {
            Object.keys(errorInfo).forEach(key => {
              scope.setExtra(key, errorInfo[key])
            })
            Sentry.captureException(error)
          })
        }
      })
    : lifecycle<Props, {}, {}>({
        componentDidCatch(error) {
          this.props.setErrors(error)
        }
      })
)

export default sentry((props: Props) => {
  const { children, errors } = props

  return (
    <Provider store={store}>
      <ConnectedRouter history={history}>
        {errors ? <ErrorPage /> : children}
      </ConnectedRouter>
    </Provider>
  )
})
