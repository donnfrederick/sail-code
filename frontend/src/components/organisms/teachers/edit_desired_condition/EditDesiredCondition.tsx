import LangLevelSelect from 'components/molecules/teachers/LangLevelSelect'
import { Info } from 'models/teachers'
import * as React from 'react'

interface Props {
  info: Info
  register(info: Info): void
}

export default (props: Props) => {
  const { info, register } = props
  return <LangLevelSelect info={info} register={register} />
}
