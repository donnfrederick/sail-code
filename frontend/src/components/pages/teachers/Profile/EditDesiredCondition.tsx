import TeachersProfileEditDesiredCondition from 'components/templates/teachers/profile/EditDesiredCondition'
import * as React from 'react'

export default () => {
  return <TeachersProfileEditDesiredCondition />
}
