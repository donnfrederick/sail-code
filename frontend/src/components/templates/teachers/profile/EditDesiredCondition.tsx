import EditPageTitle from 'components/atoms/teachers/EditPageTitle'
import Header from 'components/molecules/teachers/Header'
import EditDesiredCondition from 'components/organisms/teachers/edit_desired_condition'
import EditPageButtons from 'components/organisms/teachers/edit_page_buttons'
import * as React from 'react'
import styled from 'styled-components'

export default () => {
  return (
    <Container>
      <EditPageTitle
        text={'新しい希望条件を\n選んでください'}
        marginBottom={180}
      />
      <EditDesiredCondition />
      <EditPageButtons type="desired_condition" />
      <Header
        hasSupport={true}
        text="学生への希望条件を変更"
        backToHome={true}
      />
    </Container>
  )
}

const Container = styled.div`
  position: relative;
  width: 100%;
  min-height: calc(var(--vh, 1vh) * 100);
  box-sizing: border-box;
  padding: 192px 0 56px;
  text-align: center;
`
