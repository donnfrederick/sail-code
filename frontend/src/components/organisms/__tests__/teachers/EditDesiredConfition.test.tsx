import Router from 'components/organisms/Router'
import EditDesiredCondition from 'components/organisms/teachers/edit_desired_condition'
import * as React from 'react'
import * as renderer from 'react-test-renderer'

test('render <EditDesiredCondition />', () => {
  const tree = renderer
    .create(
      <Router>
        <EditDesiredCondition />
      </Router>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
