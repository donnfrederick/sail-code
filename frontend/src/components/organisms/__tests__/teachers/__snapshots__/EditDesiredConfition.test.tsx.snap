// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render <EditDesiredCondition /> 1`] = `
<div
  className="sc-gzVnrw dUjKSS"
>
  <button
    className="sc-EHOje bpREwu"
    color="#138efd"
    data-disabled={false}
    data-selected={false}
    fontSize={36}
    height={100}
    onClick={[Function]}
    width={640}
  >
    日本語が得意な方がいい
  </button>
  <button
    className="sc-EHOje bpREwu"
    color="#138efd"
    data-disabled={false}
    data-selected={false}
    fontSize={36}
    height={100}
    onClick={[Function]}
    width={640}
  >
    日本語が不得意でも構わない
  </button>
</div>
`;
